zebra-chain/src/orchard/address.rs

```diff
@@ -0,0 +1,87 @@
+//! Orchard shielded payment addresses.
+
+use std::fmt;
+
+#[cfg(test)]
+use proptest::prelude::*;
+
+use crate::parameters::Network;
+
+use super::keys;
+
+/// A Orchard _shielded payment address_.
+///
+/// Also known as a _diversified payment address_ for Orchard, as
+/// defined in [§5.6.4.1 of the Zcash Specification][orchardpaymentaddrencoding].
+///
+/// [orchardpaymentaddrencoding]: https://zips.z.cash/protocol/nu5.pdf#orchardpaymentaddrencoding
+#[derive(Clone, Copy, Eq, PartialEq)]
+pub struct Address {
+    network: Network,
+    diversifier: keys::Diversifier,
+    transmission_key: keys::TransmissionKey,
+}
+
+impl fmt::Debug for Address {
+    fn fmt(&self, f: &mut fmt::Formatter) -> fmt::Result {
+        f.debug_struct("OrchardAddress")
+            .field("network", &self.network)
+            .field("diversifier", &self.diversifier)
+            .field("transmission_key", &self.transmission_key)
+            .finish()
+    }
+}
+
+#[cfg(test)]
+impl Arbitrary for Address {
```

Comment:
Can you move Address to orchard::arbitrary?

That way, it will be available using the proptest-impl feature as well.